### ISSUE TITLE
'stddef.h' file not found on v1.15.1

### ISSUE THREAD

**Issue by KagaJiankui** [microsoft/vscode-cpptools#10733]
### Environment

- OS and Version: Windows 11 22H2
- VS Code Version: 1.76.2
- C/C++ Extension Version: 1.15.1 (Pre-release)
- If using SSH remote, specify OS of remote machine:  Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 GNU/Linux

### Bug Summary and Steps to Reproduce

Bug Summary:

Steps to reproduce:
1. Open a C++ file
2. See the `'stddef.h' file not found` error in the `include` field
![image](https://user-images.githubusercontent.com/58453886/227071401-ae7fd490-3138-4055-bd84-4ab73344637b.png)

Expected behavior:
This error should be fixed since version 1.13.4, and on version 1.14.3 this bug is never emitted.

**User-wide Configuration**
The following JSON code is all the user-wide configuration related to C/C++ extension.

```json
"C_Cpp.autocompleteAddParentheses": true,
    "C_Cpp.clang_format_fallbackStyle": "LLVM",
    "C_Cpp.clang_format_path": "C:\\Program Files\\LLVM\\bin\\clang-format.exe",
    "C_Cpp.clang_format_sortIncludes": true,
    "C_Cpp.codeAnalysis.clangTidy.args": [
        // "--fix",
        // "--fix-errors",
        // "--fix-notes",
        "--use-color"
    ],
    "C_Cpp.codeAnalysis.clangTidy.enabled": true,
    "C_Cpp.codeAnalysis.clangTidy.useBuildPath": true,
    "C_Cpp.default.cStandard": "c11",
    "C_Cpp.default.compilerArgs": [],
    "C_Cpp.default.compilerPath": "",
    "C_Cpp.default.cppStandard": "c++11",
    "C_Cpp.default.includePath": [
        "${default}"
    ],
    "C_Cpp.default.intelliSenseMode": "windows-msvc-x64",
    "C_Cpp.default.windowsSdkVersion": "10.0.18362.0",
    "C_Cpp.experimentalFeatures": "enabled",
    "C_Cpp.formatting": "clangFormat",
    "C_Cpp.inlayHints.autoDeclarationTypes.showOnLeft": true,
    "C_Cpp.intelliSenseCacheSize": 6144,
    "C_Cpp.intelliSenseEngineFallback": "enabled",
    "C_Cpp.intelliSenseUpdateDelay": 1000,
    // "timeline.showView": true,
    "C_Cpp.vcFormat.indent.caseLabels": true,
    "C_Cpp.vcFormat.newLine.closeBraceSameLine.emptyFunction": true,
    "C_Cpp.vcFormat.newLine.closeBraceSameLine.emptyType": true
```

### Configuration and Logs

```shell
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${workspaceFolder}/include/**"
            ],
            "defines": [],
            "compilerPath": "/usr/bin/clang",
            "intelliSenseMode": "linux-clang-x64",
            "configurationProvider": "ms-vscode.cmake-tools",
            "compileCommands": "${workspaceFolder}/build/compile_commands.json"
        }
    ],
    "version": 4
}

Language Server Logging:
loggingLevel: Debug
loggingLevel 已更改为: Debug
LSP: cpptools/getDocumentSymbols: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 265)
LSP: cpptools/getDocumentSymbols: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 266)
LSP: $/cancelRequest (cpptools/getDocumentSymbols, id: 265)
LSP: cpptools/textEditorSelectionChange
LSP: cpptools/textEditorSelectionChange
LSP: cpptools/getDocumentSymbols: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 267)
LSP: cpptools/getCodeActions: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 268)
LSP: cpptools/activeDocumentChange: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp
LSP: cpptools/getCodeActions: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 269)
LSP: textDocument/hover: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 270)
LSP: textDocument/hover: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 271)
LSP: textDocument/hover: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 272)
LSP: textDocument/hover: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 273)
LSP: cpptools/getCodeActions: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 274)
LSP: textDocument/hover: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 275)
LSP: cpptools/getCodeActions: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 276)
LSP: cpptools/queryTranslationUnitSource: file:///usr/include/stdlib.h (id: 277)
LSP: cpptools/textEditorSelectionChange
LSP: cpptools/textEditorSelectionChange
LSP: textDocument/didOpen: file:///usr/include/stdlib.h
LSP: cpptools/getCodeActions: file:///usr/include/stdlib.h (id: 278)
LSP: cpptools/activeDocumentChange: file:///usr/include/stdlib.h
LSP: cpptools/getDocumentSymbols: file:///usr/include/stdlib.h (id: 279)
LSP: cpptools/getDocumentSymbols: file:///usr/include/stdlib.h (id: 280)
LSP: cpptools/getSemanticTokens: file:///usr/include/stdlib.h (id: 281)
LSP: cpptools/resumeParsing
LSP: cpptools/getDocumentSymbols: file:///usr/include/stdlib.h (id: 282)
LSP: $/cancelRequest (cpptools/getDocumentSymbols, id: 282)
正在检查语法错误: /usr/include/stdlib.h
正在为以下项的翻译单元中文件的 IntelliSense 更新进行排队: /home/lexington/CS205_2022F/project2/src/vector_dot.cpp
LSP: cpptools/getCodeActions: file:///usr/include/stdlib.h (id: 283)
错误波形曲线计数: 0
错误波形曲线计数: 0
错误波形曲线计数: 0
更新 IntelliSense 时间(秒): 0.306
LSP: cpptools/getSemanticTokens: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 284)
LSP: cpptools/getSemanticTokens: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 285)
LSP: cpptools/getInlayHints: file:///usr/include/stdlib.h (id: 286)
LSP: textDocument/hover: file:///usr/include/stdlib.h (id: 287)
LSP: cpptools/getFoldingRanges: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 288)
LSP: textDocument/hover: file:///usr/include/stdlib.h (id: 289)
LSP: textDocument/hover: file:///usr/include/stdlib.h (id: 290)
LSP: cpptools/getCodeActions: file:///usr/include/stdlib.h (id: 291)
LSP: textDocument/hover: file:///usr/include/stdlib.h (id: 292)
LSP: textDocument/hover: file:///usr/include/stdlib.h (id: 293)
LSP: textDocument/hover: file:///usr/include/stdlib.h (id: 294)
LSP: textDocument/documentHighlight: file:///usr/include/stdlib.h (id: 295)
LSP: cpptools/textEditorSelectionChange
LSP: cpptools/getFoldingRanges: file:///usr/include/stdlib.h (id: 296)
LSP: cpptools/getCodeActions: file:///usr/include/stdlib.h (id: 297)
LSP: textDocument/definition: file:///usr/include/stdlib.h (id: 298)
LSP: cpptools/queryTranslationUnitSource: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h (id: 299)
LSP: cpptools/textEditorSelectionChange
LSP: textDocument/didOpen: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h
LSP: cpptools/getCodeActions: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h (id: 300)
LSP: cpptools/activeDocumentChange: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h
LSP: cpptools/getDocumentSymbols: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h (id: 301)
LSP: cpptools/getSemanticTokens: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h (id: 302)
LSP: cpptools/getInlayHints: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h (id: 303)
LSP: cpptools/getDocumentSymbols: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h (id: 304)
正在检查语法错误: /usr/lib/llvm-17/lib/clang/17/include/stddef.h
正在为以下项的翻译单元中文件的 IntelliSense 更新进行排队: /home/lexington/CS205_2022F/project2/src/vector_dot.cpp
错误波形曲线计数: 0
错误波形曲线计数: 0
错误波形曲线计数: 0
更新 IntelliSense 时间(秒): 0.28
LSP: cpptools/getSemanticTokens: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 305)
LSP: cpptools/getSemanticTokens: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 306)
LSP: textDocument/hover: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h (id: 307)
LSP: textDocument/hover: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h (id: 308)
LSP: cpptools/getFoldingRanges: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h (id: 309)
LSP: cpptools/getFoldingRanges: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 310)
LSP: cpptools/didChangeCppProperties (id: 311)
LSP: cpptools/didChangeCppProperties (id: 312)
LSP: cpptools/didChangeCppProperties (id: 313)
LSP: cpptools/didChangeCppProperties (id: 314)
LSP: cpptools/textEditorSelectionChange
LSP: cpptools/textEditorSelectionChange
LSP: cpptools/getDocumentSymbols: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 315)
LSP: cpptools/getCodeActions: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 316)
LSP: cpptools/activeDocumentChange: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp
LSP: cpptools/resumeParsing
LSP: cpptools/getInlayHints: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 317)
LSP: cpptools/getCodeActions: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 318)
空闲环路: 重新分析活动文档
正在检查语法错误: /home/lexington/CS205_2022F/project2/src/vector_dot.cpp
正在为以下项的翻译单元中文件的 IntelliSense 更新进行排队: /home/lexington/CS205_2022F/project2/src/vector_dot.cpp
LSP: textDocument/hover: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 319)
错误波形曲线计数: 0
错误波形曲线计数: 0
错误波形曲线计数: 0
更新 IntelliSense 时间(秒): 0.268
LSP: cpptools/getSemanticTokens: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 320)
LSP: cpptools/getSemanticTokens: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 321)
LSP: cpptools/getSemanticTokens: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h (id: 322)
LSP: textDocument/hover: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 323)
LSP: cpptools/getCodeActions: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 324)
LSP: cpptools/getFoldingRanges: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 325)
LSP: cpptools/getFoldingRanges: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 326)
LSP: textDocument/hover: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 327)
LSP: textDocument/hover: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 328)
LSP: cpptools/getCodeActions: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 329)
LSP: textDocument/hover: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 330)
LSP: cpptools/getCodeActions: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 331)
LSP: cpptools/getInlayHints: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 332)
LSP: textDocument/didClose: file:///usr/lib/llvm-17/lib/clang/17/include/stddef.h

Command palette logs:
-------- Diagnostics - 3/23/2023, 8:57:52 AM
Version: 1.15.1
Current Configuration:
{
    "name": "Linux",
    "includePath": [
        "/home/lexington/CS205_2022F/project2/include/**"
    ],
    "defines": [],
    "compilerPath": "/usr/bin/clang",
    "intelliSenseMode": "linux-clang-x64",
    "configurationProvider": "ms-vscode.cmake-tools",
    "compileCommands": "/home/lexington/CS205_2022F/project2/build/compile_commands.json",
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true,
    "rawCompilerPath": "/usr/bin/clang",
    "cStandard": "c11",
    "cppStandard": "c++11",
    "mergeConfigurations": false,
    "browse": {
        "path": [
            "${workspaceFolder}/include/**",
            "${workspaceFolder}"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Custom browse configuration: 
{
    "browsePath": [
        "/home/lexington/CS205_2022F/project2/include",
        "/home/lexington/CS205_2022F/project2/src"
    ],
    "compilerPath": "clang++",
    "compilerArgs": [],
    "compilerFragments": [
        " -O0 -g -g"
    ]
}
Custom configurations:
[ /home/lexington/CS205_2022F/project2/src/vector_dot.cpp ]
{
    "includePath": [
        "/home/lexington/CS205_2022F/project2/include"
    ],
    "defines": [],
    "compilerPath": "clang++",
    "compilerArgs": [],
    "compilerFragments": [
        " -O0 -g -g"
    ]
}
Translation Unit Mappings:
[ /home/lexington/CS205_2022F/project2/src/vector_dot.cpp ]:
    /home/lexington/CS205_2022F/project2/src/vector_dot.cpp
    /home/lexington/CS205_2022F/project2/include/vector_dot.h *
    /usr/include/stdlib.h *
Translation Unit Configurations:
[ /home/lexington/CS205_2022F/project2/src/vector_dot.cpp ]:
    Process ID: 1950
    Memory Usage: 143 MB
    Compiler Path: /usr/bin/clang++
    Includes:
        /home/lexington/CS205_2022F/project2/include
        /usr/include/c++/11
        /usr/include/x86_64-linux-gnu/c++/11
        /usr/include/c++/11/backward
        /usr/lib/llvm-17/lib/clang/17/include
        /usr/local/include
        /usr/include/x86_64-linux-gnu
        /usr/include
    Standard Version: c++17
    IntelliSense Mode: linux-clang-x64
    Other Flags:
        --clang
        --clang_version=170000
Total Memory Usage: 143 MB
Browse Paths from compile_commands.json, from workspace folder: /home/lexington/CS205_2022F/project1
    /home/lexington/CS205_2022F/project1/include
    /home/lexington/CS205_2022F/project1/src
Browse Paths from compile_commands.json, from workspace folder: /home/lexington/CS205_2022F/project2
    /home/lexington/CS205_2022F/project2/include
    /home/lexington/CS205_2022F/project2/src

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 43911
Number of files parsed: 534
```

**Comment by sean-mcmanus** [comment]
The error is coming from clang-tidy. If you set C_Cpp.loggingLevel to debug, we output the clang-tidy command line we use to the C/C++ output window. You normally should see something like
`-isystem/usr/lib/llvm-17/lib/clang/17/include` at the end of the logging and the stddef.h would be found from there. If you use Go to Definition on the stddef.h in the code line `#incluede <stddef.h>` does it open a file that is part of a `-isystem` passed to clang-tidy?

The only change we made to this area for 1.15.x is we stopped automatically sending `-mno-sse2`, so you could try using
```json
    "C_Cpp.codeAnalysis.clangTidy.args": [
        "--extra-arg=-mno-sse2"
    ],
```

Are you able to find out why clang-tidy is not finding stddef.h?

**Comment by KagaJiankui** [comment]
This is the debug logging when the `C_Cpp.loggingLevel` is set to debug. There seems to be nothing related to the `-isystem`.

``` log /home/lexington/.vscode-server/extensions/ms-vscode.cpptools-1.15.1-linux-x64/bin/../LLVM/bin/clang-tidy
--use-color
--extra-arg=-mno-sse2
--export-fixes=/tmp/{13461097544554508563}/fixes140524241528384.yaml
--quiet
--use-color=false
/home/lexington/CS205_2022F/project2/src/vector_dot.cpp
-p=/home/lexington/CS205_2022F/project2/build/
LSP: cpptools/queryTranslationUnitSource: file:///home/lexington/CS205_2022F/project2/src/vector_dot.cpp (id: 152)
LSP: cpptools/finishedRequestCustomConfig: /home/lexington/CS205_2022F/project2/src/vector_dot.cpp
LSP: cpptools/getDocumentSymbols: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 153)
LSP: cpptools/getDocumentSymbols: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 154)
LSP: $/cancelRequest (cpptools/getDocumentSymbols, id: 153)
Error while processing /home/lexington/CS205_2022F/project2/src/vector_dot.cpp.
/home/lexington/CS205_2022F/project2/src/vector_dot.cpp:10:24: warning: narrowing conversion from 'unsigned long' to signed type 'int' is implementation-defined [cppcoreguidelines-narrowing-conversions]
  memset(vec->data, 0, sizeof(double) * size);
                       ^
/home/lexington/CS205_2022F/project2/src/vector_dot.cpp:17:27: warning: narrowing conversion from 'unsigned long' to signed type 'int' is implementation-defined [cppcoreguidelines-narrowing-conversions]
  memcpy(vec->data, data, sizeof(double) * size);
                          ^
/home/lexington/CS205_2022F/project2/src/vector_dot.cpp:38:22: warning: narrowing conversion from 'unsigned long' to signed type 'int' is implementation-defined [cppcoreguidelines-narrowing-conversions]
    memcpy(dst, src, sizeof(double) * src->length);
                     ^
/home/lexington/CS205_2022F/project2/src/vector_dot.cpp:46:22: warning: narrowing conversion from 'unsigned long' to signed type 'int' is implementation-defined [cppcoreguidelines-narrowing-conversions]
    memcpy(dst, src, sizeof(double) * src->length);
                     ^
/home/lexington/CS205_2022F/project2/src/vector_dot.cpp:58:25: warning: narrowing conversion from 'unsigned long' to signed type 'int' is implementation-defined [cppcoreguidelines-narrowing-conversions]
  memcpy(vec, data_new, sizeof(double) * size_new);
                        ^
/usr/include/stdlib.h:32:10: error: 'stddef.h' file not found [clang-diagnostic-error]
#include <stddef.h>
         ^~~~~~~~~~
LSP: cpptools/getFoldingRanges: file:///home/lexington/CS205_2022F/project2/include/vector_dot.h (id: 155)
```

**Comment by KagaJiankui** [comment]
after disabling `C_Cpp.codeAnalysis.clangTidy.useBuildPath` and adding the `--extra-arg=-mno-sse2` flag to clang-tidy, the error stopped to emit.
